src/pip/_internal/utils/compat.py

```diff
@@ -121,10 +121,8 @@ def str_to_display(data, desc=None):
     try:
         decoded_data = data.decode(encoding)
     except UnicodeDecodeError:
-        if desc is None:
-            desc = 'Bytes object'
-        msg_format = '{} does not appear to be encoded as %s'.format(desc)
```

Comment:
Nit: Perhaps we can update this to 

```python
msg_format = '{} does not appear to be encoded as %s'.format(desc or 'Bytes Object')
```

instead of doing it in `logger.warning`

Reply:
Direct call to `str.format` was avoided to make this patch the best practice we can get as suggested by @uranusjr.  As for the `desc or 'Bytes object'`, we can't fit it in one line even if it was just `desc` so I kept it that way.  I don't really have a strong opinion on this though.

Reply:
Yes, it’s easier for debugging if the logger can get the data as structured as possible, and each format call loses a bit of structural context. For the same reason, I’m fine with either the current form or setting `desc = 'Bytes object'` before the logging call, since both pass the same thing to the logger.